lib/apis/moray.js

```diff
@@ -586,8 +596,25 @@ Moray.prototype.putVm = function (uuid, vm, cb) {
     this.connection.putObject(VMS_BUCKET_NAME, uuid, object, cb);
 };
 
+/*
+ * Deletes *all* VMs that match the filter generated from "params".
+ */
```

Comment:
What do you mean by:

> we don't seem to actually allow record deletion for VMs

? If you mean that VMs are just deleted from the `vmapi_vms` Moray bucket and do not go through the VMs deletion workflow, that's correct. But we don't just mark them as destroyed, they are removed from moray.

> It looks like this method is really just for the test harness; is that correct?

That's correct.

I had initially put it in `test/lib`, and then wanted to delete VMs with specific properties. That's why I needed to use `Moray.prototype_createSearch` and thought it would actually make sense to add `delVms` to `Moray.prototype`. 

I could also put `delVms` back in `test/lib` if leaving it on `Moray.prototype` is too confusing.